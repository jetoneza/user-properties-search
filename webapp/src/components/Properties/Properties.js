import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Card, Icon } from 'semantic-ui-react';

function Properties({ properties }) {
  return (
    <Grid centered stackable columns={3}>
      { properties.map((property) => {
        return (
          <Grid.Column key={property.id}>
            <Card>
              <Card.Content>
                <Card.Header>{ property.street }</Card.Header>
                <Card.Meta>
                  <span>{ property.city }, { property.state }</span>
                </Card.Meta>
                <Card.Description>${ property.rent }</Card.Description>
              </Card.Content>
              <Card.Content extra>
                <a>
                  <Icon name='user' />
                  { property.user.firstName } { property.user.lastName }
                </a>
              </Card.Content>
            </Card>
          </Grid.Column>
        );
      }) }
    </Grid>
  );
}

Properties.propTypes = {
  properties: PropTypes.arrayOf(
    PropTypes.shape({
      street: PropTypes.string.isRequired,
      city: PropTypes.string.isRequired,
      state: PropTypes.string.isRequired,
      rent: PropTypes.number.isRequired,
      user: PropTypes.shape({
        firstName: PropTypes.string.isRequired,
        lastName: PropTypes.string.isRequired,
      }),
    })
  ).isRequired,
}

export default Properties;
