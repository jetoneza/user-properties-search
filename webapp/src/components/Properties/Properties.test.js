import React from 'react';
import renderer from 'react-test-renderer';
import { Card } from 'semantic-ui-react';

import { shallow } from 'enzyme';

const properties = [
  {
    "id": "11",
    "street": "Wintheiser Gateway",
    "city": "Samirstad",
    "state": "South Carolina",
    "zip": "67045",
    "rent": 829.6,
    "user": {
      "id": "54",
      "firstName": "Lukas",
      "lastName": "Huels"
    }
  },
  {
    "id": "12",
    "street": "Braulio Cape",
    "city": "North Kristin",
    "state": "Vermont",
    "zip": "33846-6770",
    "rent": 629.33,
    "user": {
      "id": "17",
      "firstName": "Liliana",
      "lastName": "Wehner"
    }
  },
  {
    "id": "13",
    "street": "Javier Groves",
    "city": "Port Mallieton",
    "state": "Indiana",
    "zip": "44469-6898",
    "rent": 451.02,
    "user": {
      "id": "91",
      "firstName": "Arvid",
      "lastName": "Bailey"
    }
  },
];

import Properties from './Properties';

describe('<Properties /> test', () => {
  const props = {
    properties,
  };

  test('Should render list', () => {
    const wrapper = shallow(<Properties {...props}/>);

    expect(wrapper.find(Card).length).toEqual(3);
  });

  test('Should render login page', () => {
    const component = <Properties {...props}/>
      shallow(component);

    const renderedComponent = renderer.create(component);
    const tree = renderedComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
