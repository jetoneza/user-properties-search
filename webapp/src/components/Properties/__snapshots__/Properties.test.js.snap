// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Properties /> test Should render login page 1`] = `
<div
  className="ui centered stackable three column grid"
>
  <div
    className="column"
  >
    <div
      className="ui card"
      href={undefined}
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="header"
        >
          Wintheiser Gateway
        </div>
        <div
          className="meta"
        >
          <span>
            Samirstad
            , 
            South Carolina
          </span>
        </div>
        <div
          className="description"
        >
          $
          829.6
        </div>
      </div>
      <div
        className="extra content"
      >
        <a>
          <i
            aria-hidden="true"
            className="user icon"
          />
          Lukas
           
          Huels
        </a>
      </div>
    </div>
  </div>
  <div
    className="column"
  >
    <div
      className="ui card"
      href={undefined}
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="header"
        >
          Braulio Cape
        </div>
        <div
          className="meta"
        >
          <span>
            North Kristin
            , 
            Vermont
          </span>
        </div>
        <div
          className="description"
        >
          $
          629.33
        </div>
      </div>
      <div
        className="extra content"
      >
        <a>
          <i
            aria-hidden="true"
            className="user icon"
          />
          Liliana
           
          Wehner
        </a>
      </div>
    </div>
  </div>
  <div
    className="column"
  >
    <div
      className="ui card"
      href={undefined}
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="header"
        >
          Javier Groves
        </div>
        <div
          className="meta"
        >
          <span>
            Port Mallieton
            , 
            Indiana
          </span>
        </div>
        <div
          className="description"
        >
          $
          451.02
        </div>
      </div>
      <div
        className="extra content"
      >
        <a>
          <i
            aria-hidden="true"
            className="user icon"
          />
          Arvid
           
          Bailey
        </a>
      </div>
    </div>
  </div>
</div>
`;
